src/Microsoft.ML.Data/DataView/DetectAnomalyBySrCnnBatchTransform.cs

```diff
@@ -0,0 +1,332 @@
+                    return _input.GetGetter<TValue>(_input.Schema[col]);
+                }
+
+                Ch.AssertValue(_getters);
+                var getter = _getters[col];
+                Ch.Assert(getter != null);
+                var fn = getter as ValueGetter<TValue>;
+                if (fn == null)
+                    throw Ch.Except("Invalid TValue in GetGetter: '{0}'", typeof(TValue));
+                return fn;
+            }
+
+            public override ValueGetter<DataViewRowId> GetIdGetter()
+            {
+                return
+                    (ref DataViewRowId val) =>
+                    {
+                        Ch.Check(IsGood, "Cannot call ID getter in current state");
+                        val = new DataViewRowId((ulong)Position, 0);
+                    };
+            }
+
+            public override bool IsColumnActive(DataViewSchema.Column column)
+            {
+                Ch.Check(column.Index < _parent.SchemaBindings.AsSchema.Count);
+                return _active[column.Index];
+            }
+
+            protected override bool MoveNextCore()
+            {
+                if (!_input.MoveNext())
+                    return false;
+                if (!_firstInBatchInInputCursorDel())
+                    return true;
+
+                // If we are here, this means that _input.MoveNext() has gotten us to the beginning of the next batch,
+                // so now we need to look ahead at the entire next batch in the _lookAheadCursor.
+                // The _lookAheadCursor's position should be on the last row of the previous batch (or -1).
+                Ch.Assert(_lastInBatchInLookAheadCursorDel());
+
+                var good = _lookAheadCursor.MoveNext();
+                // The two cursors should have the same number of elements, so if _input.MoveNext() returned true,
+                // then it must return true here too.
+                Ch.Assert(good);
+
+                do
+                {
+                    _inputGetterInLookAheadCursor(ref _currentInput);
+                    _parent.ProcessExample(_currentBatch, _currentInput);
+                } while (!_lastInBatchInLookAheadCursorDel() && _lookAheadCursor.MoveNext());
+
+                _parent.ProcessBatch(_currentBatch);
+                return true;
+            }
+        }
+    }
+
+    // TODO: SrCnn
+    public sealed class DetectAnomalyBySrCnnBatchTransform : BatchTransformBase<float, DetectAnomalyBySrCnnBatchTransform.Batch>
+    {
+        private readonly int _batchSize;
+
+        public DetectAnomalyBySrCnnBatchTransform(IHostEnvironment env, IDataView input, string inputColumnName, string outputColumnName, double threshold, int batchSize, double sensitivity, DetectMode detectMode)
+            : base(env, input, inputColumnName, outputColumnName, new VectorDataViewType(NumberDataViewType.Double, batchSize))
+        {
+            _batchSize = batchSize;
```

Comment:
>batchSize [](start = 25, length = 9)

Check that `batchSize > 0` (and if there is an upper bound to it then that as well).

Reply:
Sure. This is just the a draft API PR, but I have added it anyway.